src/meta/processors/admin/BalancePlan.cpp

```diff
@@ -141,37 +142,38 @@ bool BalancePlan::recovery(bool resume) {
             return false;
         }
         while (iter->valid()) {
-            BalanceTask task;
-            task.kv_ = kv_;
-            task.client_ = client_;
+            auto task = std::make_unique<BalanceTask>();
+            task->kv_ = kv_;
+            task->client_ = client_;
             {
                 auto tup = BalanceTask::parseKey(iter->key());
-                task.balanceId_ = std::get<0>(tup);
-                task.spaceId_ = std::get<1>(tup);
-                task.partId_ = std::get<2>(tup);
-                task.src_ = std::get<3>(tup);
-                task.dst_ = std::get<4>(tup);
-                task.taskIdStr_ = task.buildTaskId();
+                task->balanceId_ = std::get<0>(tup);
+                task->spaceId_ = std::get<1>(tup);
+                task->partId_ = std::get<2>(tup);
+                task->src_ = std::get<3>(tup);
+                task->dst_ = std::get<4>(tup);
+                task->taskIdStr_ = task->buildTaskId();
             }
             {
                 auto tup = BalanceTask::parseVal(iter->val());
-                task.status_ = std::get<0>(tup);
-                task.ret_ = std::get<1>(tup);
-                task.srcLived_ = std::get<2>(tup);
-                task.startTimeMs_ = std::get<3>(tup);
-                task.endTimeMs_ = std::get<4>(tup);
-                if (resume && task.ret_ != BalanceTask::Result::SUCCEEDED) {
-                    // Resume the failed task.
-                    task.ret_ = BalanceTask::Result::IN_PROGRESS;
-                    task.status_ = BalanceTask::Status::START;
-                    if (ActiveHostsMan::isLived(kv_, task.src_)) {
-                        task.srcLived_ = true;
+                task->status_ = std::get<0>(tup);
+                task->ret_ = std::get<1>(tup);
+                task->srcLived_ = std::get<2>(tup);
+                task->startTimeMs_ = std::get<3>(tup);
+                task->endTimeMs_ = std::get<4>(tup);
+                if (resume && task->ret_ != BalanceTask::Result::SUCCEEDED) {
+                    // Resume the failed or stopped task
+                    task->ret_ = BalanceTask::Result::IN_PROGRESS;
+                    task->status_ = BalanceTask::Status::START;
+                    if (ActiveHostsMan::isLived(kv_, task->src_)) {
+                        task->srcLived_ = true;
                     } else {
-                        task.srcLived_ = false;
+                        task->srcLived_ = false;
                     }
```

Comment:
Would better to simplify the code?
task->srcLived_ = ActiveHostsMan::isLived(kv_, task->src_);

Reply:
Good suggestion